src/components/views/ViewReusablePaymentRequest.tsx

```diff
@@ -1,44 +1,79 @@
 import { usePaymentRequests } from '@/hooks/cashu/usePaymentRequests';
 import QRCode from 'qrcode.react';
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import ClipboardButton from '../buttons/utility/ClipboardButton';
 import { useCashuContext } from '@/hooks/contexts/cashuContext';
 import { decodePaymentRequest } from '@cashu/cashu-ts';
 import useMintlessMode from '@/hooks/boardwalk/useMintlessMode';
 import { Spinner } from 'flowbite-react';
+import { usePolling } from '@/hooks/util/usePolling';
+import { useCashu } from '@/hooks/cashu/useCashu';
 
-const ViewReusablePaymentRequest = () => {
-   const [paymentRequest, setPaymentRequest] = useState('');
-   const { fetchPaymentRequest, fetchingPaymentRequest } = usePaymentRequests();
+interface ViewReusablePaymentRequestProps {
+   onSuccess?: () => void;
+}
+
+const ViewReusablePaymentRequest = ({ onSuccess }: ViewReusablePaymentRequestProps) => {
+   const { fetchPaymentRequest, lookupLastPaid } = usePaymentRequests();
+   const [paymentRequest, setPaymentRequest] = useState<{ pr: string; id: string } | null>(null);
+   const [lastPaid, setLastPaid] = useState<string | null>(null);
    const { activeWallet, activeUnit } = useCashuContext();
+   const { handleClaimToSourceMint } = useCashu();
    const { isMintless } = useMintlessMode();
 
+   const pollPaymentRequest = useCallback(async () => {
+      if (!paymentRequest?.id) return;
+
+      try {
+         const res = await lookupLastPaid(paymentRequest.id);
+
+         if (!lastPaid) {
+            /* First time checking - initialize lastPaid */
+            setLastPaid(res.lastPaid || new Date(0).toISOString());
+            return;
+         }
+
+         if (res.lastPaid && res.lastPaid > lastPaid) {
+            /* payment request was paid since this component has been open */
+            handleClaimToSourceMint(res.token);
+            onSuccess?.();
+         }
+      } catch (e) {
+         console.warn('Failed to poll payment request:', e);
+      }
+   }, [paymentRequest?.id, lastPaid, lookupLastPaid, handleClaimToSourceMint, onSuccess]);
+
+   usePolling(pollPaymentRequest, 1000);
+
    useEffect(() => {
       if (!activeWallet) return;
       if (!activeUnit) return;
       const storedPaymentRequest = localStorage.getItem('paymentRequest');
-      if (storedPaymentRequest) {
+      const storedRequestId = localStorage.getItem('paymentRequestId');
```

Comment:
is this id id of the object stored in `paymentRequest`? if yes why do we store them separately?

Reply:
yea, refactored it to use the same localstorage key